src/rrdp.rs

```diff
@@ -361,95 +463,234 @@ pub trait ProcessDelta {
+            let first = s.next().ok_or_else(
+                ParseHashError::bad_chars
+            )?.to_digit(16).ok_or_else(
+                ParseHashError::bad_chars
+            )?;
+            let second = s.next().ok_or_else(
+                ParseHashError::bad_chars
+            )?.to_digit(16).ok_or_else(
+                ParseHashError::bad_chars
+            )?;
+            *octet = (first << 4 | second) as u8;
         }
-        Ok(DigestHex(res))
+        Ok(Hash(res))
     }
 }
 
-impl ops::Deref for DigestHex {
-    type Target = [u8];
 
-    fn deref(&self) -> &[u8] {
+//--- AsRef
+
+impl AsRef<[u8]> for Hash {
+    fn as_ref(&self) -> &[u8] {
         self.0.as_ref()
     }
 }
 
-impl AsRef<[u8]> for DigestHex {
-    fn as_ref(&self) -> &[u8] {
-        self.0.as_ref()
+
+//--- PartialEq
+//
+// PartialEq<Self> and Eq are derived.
+
+impl PartialEq<digest::Digest> for Hash {
+    fn eq(&self, other: &digest::Digest) -> bool {
+        // XXX This doesn’t properly check the algorithm.
+        self.0.as_ref() == other.as_ref()
     }
 }
 
-impl fmt::Display for DigestHex {
+
+//--- Display and Debug
+
+impl fmt::Display for Hash {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        for &ch in self.0.as_slice() {
+        for &ch in self.as_slice() {
             write!(f, "{:02x}", ch)?;
         }
         Ok(())
     }
 }
 
+impl fmt::Debug for Hash {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Hash({})", self)
+    }
+}
+
 
 //------------ Action --------------------------------------------------------
 
+/// The choice of actions in a delta file.
 enum Action {
+    /// An object is to be inserted or updated.
+    ///
+    /// The object is to be updated, if a hash is given. Otherwise it is
+    /// to be inserted.
     Publish,
+
+    /// An object is to be deleted.
     Withdraw,
 }
 
 
-//------------ Xml Names -----------------------------------------------------
+//------------ ObjectReader --------------------------------------------------
+
+/// A reader providing the content of an object.
+///
+/// The content is included in base64 encoding in the RRDP’s XML. This reader
+/// provides access to the decoded data via the standard `Read` trait.
+pub struct ObjectReader<'a>(
+    /// The base64 encoded data.
+    base64::read::DecoderReader<'a, &'a [u8]>
+);
+
+impl<'a> ObjectReader<'a> {
+    /// Processes XML PCDATA as object content.
+    ///
+    /// An object reader is created and passed to the closure `op` for
+    /// actual processing.
+    fn process_text<R, T, E, F> (
+        content: &mut Content,
+        reader: &mut Reader<R>,
+        op: F
+    ) -> Result<T, E>
+    where
+        R: io::BufRead,
+        E: From<ProcessError>,
+        F: FnOnce(&mut ObjectReader) -> Result<T, E>
+    {
+        let data_b64: Vec<_> = content.take_text(reader,  |text| {
+            Ok(text.to_ascii()?.as_bytes().iter().filter_map(|b| {
```

Comment:
Is it worth an RFC reference here to say that the RRDP RFC requires the document to be US-ASCII encoded and thus why here you treat it as ASCII, or to state that the input is expected to be ASCII as required by the spec?